portfolio/src/main/webapp/script.js

```diff
@@ -15,14 +15,32 @@
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function addAndReplaceRandomFact() {
+  const facts = [
+    'I don\'t drink coffee',
+    'My birthday is on 29th of May',
+    'I love vanilla ice cream',
+    'My favourite author is Jane Austen',
+    'I love ice-skating',
+  ];
+
+  // Pick a random fact.
+  const chosenFact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  const factContainer = document.getElementById('fact-container');
+  factContainer.innerText = chosenFact;
+}
+
+function addPhoto() {
+  const imgElement = document.createElement('img');
+  // Set the url of the image
```

Comment:
I took a screenshot on some guidance on what constitutes good and bad comments:
https://screenshot.googleplex.com/rfSQPbHU5UA

Also pointing to some guidance in the C++ styleguide (this is relatively unrelated to C++):
https://google.github.io/styleguide/cppguide.html#Implementation_Comment_Donts
"Do not state the obvious. In particular, don't literally describe what code does, unless the behavior is nonobvious to a reader who understands C++ well. Instead, provide higher level comments that describe why the code does what it does, or make the code self describing."

In this particular case I could probably argue that a comment like "Set the url of the image" is pretty self explanatory, while a comment like "Remove the previous image." is genuinely useful as it's not totally clear why is innerHTML set to ''.

This is totally not a big deal, though might be useful to keep in mind whenever you're thinking about implementation comments.

Reply:
Alright, thank you, I will try to keep this in mind from no on.